magicctapipe/scripts/lst1_magic/semi_automatic_scripts/new_nsb_DL1_to_DL2.py

```diff
@@ -0,0 +1,209 @@
+
+    "Loops over all runs of all nights"
+    Nights_list = np.sort(glob.glob(f"{target_dir}/v{__version__}/{source}/DL1Stereo/*"))
+    for night in Nights_list:
+        "Night period"
+        night_date=night.split('/')[-1]
+        print('night',night_date)
+        date_lst = night_date[:4] + "_" + night_date[4:6] + "_" + night_date[6:8]
+        print(date_lst)
+        date_magic = datetime.datetime.strptime(date_lst, "%Y_%m_%d") + datetime.timedelta(days=1)
+        print('dates', date_lst, str(date_magic))
+        for p in range(len(ST_begin)):
+            if (
+                date_magic
+                >= datetime.datetime.strptime(ST_begin[p], "%Y_%m_%d")
+                ) and (
+                date_magic
+                <= datetime.datetime.strptime(ST_end[p], "%Y_%m_%d")
+                ):
+                period = ST_list[p]
+
+        Run_list = glob.glob(f"{night}/Merged/*.h5")
+        for Run in Run_list:
+            "getting the run NSB"
+            print(Run)
+            run_str=Run.split('/')[-1].split('.')[1]
+            print('run', run_str)
+            run_LST_id = run_str.lstrip('Run')
+            print('run_lst', run_LST_id)
+            nsb = df_LST[df_LST["LST1_run"] == run_LST_id]["nsb"].tolist()[0]
+            print('nsb', nsb)
+            "rounding the NSB to the nearest MC nsb value"
+            for j in range(0, len(nsb_list)-1):
+                if (nsb < nsb_limit[j + 1]) & (nsb > nsb_limit[j]):
+                    nsb = nsb_list[j]
+            "Writing on output .txt file"
+            if (nsb <= 3.1):
+                with open(f"{night}/Merged/logs/{period}_{nsb}.txt","a+") as file:
+                    file.write(f"{Run}\n")
+
+
+def bash_DL1Stereo_to_DL2(target_dir, source, env_name):
+    """
+    This function generates the bashscript for running the DL1Stereo to DL2 analisys.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    source:
+    	source name
+    env_name:
+    	conda enviroment name
+    """
+    
+    process_name = source
+    Nights_list = np.sort(glob.glob(f"{target_dir}/v{__version__}/{source}/DL1Stereo/*"))
+    for night in Nights_list:
+        File_list = glob.glob(f"{night}/Merged/logs/ST*.txt")
+        night_date=night.split('/')[-1]
+        os.makedirs(f'{target_dir}/v{__version__}/{source}/DL2/{night_date}/logs',exist_ok=True)
+        for file in File_list:
+            with open(file, "r") as f:
+                process_size = len(f.readlines()) - 1
+            if process_size < 0:
+                continue
+            nsb=file.split("/")[-1].split("_")[-1][:3]
+            p=file.split("/")[-1].split("_")[0]           
+            RFdir=f'/fefs/aswg/LST1MAGIC/mc/models/{p}/NSB{nsb}/v01.2/dec_2276/'
+            with open(f'{source}_DL1_to_DL2_{night_date}_{file.split("/")[-1].rstrip("txt")}sh', "w") as f:
+                f.write("#!/bin/sh\n\n")
+                f.write("#SBATCH -p long\n")
```

Comment:
maybe here we can switch to short (and check memory usage and job accounting) to prevent issues with IT